cpp/src/text/edit_distance.cu

```diff
@@ -137,154 +129,157 @@ struct edit_distance_matrix_levenshtein_algorithm {
 /**
  * @copydoc nvtext::edit_distance
  */
-std::unique_ptr<cudf::column> edit_distance(cudf::strings_column_view const& strings,
+std::unique_ptr<cudf::column> edit_distance(cudf::strings_column_view const& input,
                                             cudf::strings_column_view const& targets,
                                             rmm::cuda_stream_view stream,
                                             rmm::device_async_resource_ref mr)
 {
-  auto const strings_count = strings.size();
-  if (strings_count == 0) {
-    return cudf::make_empty_column(cudf::data_type{cudf::type_to_id<cudf::size_type>()});
-  }
-  if (targets.size() > 1) {
-    CUDF_EXPECTS(strings_count == targets.size(), "targets.size() must equal strings.size()");
+  auto const output_type = cudf::data_type{cudf::type_to_id<cudf::size_type>()};
+  if (input.is_empty()) { return cudf::make_empty_column(output_type); }
+  if (targets.size() != 1) {
+    CUDF_EXPECTS(input.size() == targets.size(),
+                 "targets.size() must equal input.size()",
+                 std::invalid_argument);
+  } else {
+    CUDF_EXPECTS(!targets.has_nulls(), "single target must not be null", std::invalid_argument);
   }
 
   // create device columns from the input columns
-  auto strings_column = cudf::column_device_view::create(strings.parent(), stream);
-  auto d_strings      = *strings_column;
-  auto targets_column = cudf::column_device_view::create(targets.parent(), stream);
-  auto d_targets      = *targets_column;
-
-  // calculate the size of the compute-buffer;
-  rmm::device_uvector<std::ptrdiff_t> offsets(strings_count, stream);
-  thrust::transform(rmm::exec_policy(stream),
-                    thrust::make_counting_iterator<cudf::size_type>(0),
-                    thrust::make_counting_iterator<cudf::size_type>(strings_count),
+  auto d_strings = cudf::column_device_view::create(input.parent(), stream);
+  auto d_targets = cudf::column_device_view::create(targets.parent(), stream);
+
+  // calculate the size of the compute-buffer
+  rmm::device_uvector<std::ptrdiff_t> offsets(input.size() + 1, stream);
```

Comment:
Overflow when `input.size() == INT_MAX`.

Reply:
This should not happen. `input.size()` should always be less than `max<size_type>` so +1 may equal max but not go over.